pkg/satellite/stream/stream.go

```diff
@@ -181,7 +182,36 @@ func (ss *satelliteStream) ackReceivedTelemetry(telemetryMessageAckId string) {
 	}
 }
 
+func (ss *satelliteStream) performAutoClose() {
+	log.Printf("Stream auto-close conditions met - exiting")
+	if ss.showStats {
+		metrics.logReport()
+	}
+	close(ss.closeTickerChan)
```

Comment:
It looks like the channel is closed here to trigger the `select/case` mechanism in L207. While this may work, it is making use of a side-effect of the programming language and can be puzzling. I think it would be more intuitive to explicitly write an empty struct to the channel.

Reply:
Removed the closeTickerChan so this is irrelevant now.